### DISCUSSION TITLE
Free opensource AutoGPT / BabyAGI / no OpenAI API needed / fully local installation / based on GPT4ALL is here!!!

### DISCUSSION THREAD

**Discussion by sarutobiumon** [oobabooga/text-generation-webui#1788] [General]
Please integrate with ooooooo ba boooooga webUI!

https://github.com/kroll-software/babyagi4all

The following model files have been tested successfully: (all CPU-based GGML's so you dont need to have a pawafuru GPU!)
gpt4all-lora-quantized-ggml.bin
ggml-wizardLM-7B.q4_2.bin
ggml-vicuna-7b-1.1-q4_2.bin

Demo:
https://www.youtube.com/watch?v=KlOB-3sY5gg

**Comment by Felixqian4160** [comment]
if I want to use gpu how to chang

**Comment by flurb18** [comment]
I'm working on changing it to oobas api at this moment. Should be easy!

**Comment by flurb18** [comment]
https://github.com/flurb18/babyagi4all-api it's working!

**Reply by Felixqian4160** [reply]
let me try

**Reply by Felixqian4160** [reply]
can this software run by gpu?

**Reply by flurb18** [reply]
Yep! It uses the api of a running oobabooga instance, which can use GPU if you install it correctly

**Reply by Felixqian4160** [reply]
I SEE. I RUNNING IT NOW. IF I HAVE TWO PCs, CAN I USE TO MODELS WORK WITH EACH OTHER? I SEE THE CODE, YOU delete “model", so it just can run by one model. if I use to pcs, can this two model work with each other?

**Reply by Felixqian4160** [reply]
use gpu is faster than cpu much more

**Reply by Felixqian4160** [reply]
i want let this program to save the result but it cannot

**Reply by sarutobiumon** [reply]
@oobabooga would be nice to turn this into an extension. Also would be nice is to have a list of all available extensions displayed inside the webUI like on the Automatic1111 side.

**Comment by TomLucidor** [comment]
Side note: Has anyone tried any of these platforms (and see if they can be plugged into WebUI)?
- AutoGPT (not the one from OpenAI tho) https://github.com/Significant-Gravitas/AutoGPT
- BabyAGI and TeenageAGI (simple) https://github.com/yoheinakajima/babyagi https://github.com/seanpixel/Teenage-AGI
- SuperAGI (for developers) https://github.com/TransformerOptimus/SuperAGI
- AgentGPT (having full fledged DB) https://github.com/reworkd/AgentGPT
- AGiXT https://github.com/Josh-XT/AGiXT
- ThinkGPT https://github.com/jina-ai/thinkgpt
- Spark https://github.com/cedriking/spark
